device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/https/HttpsRequest.java

```diff
@@ -33,8 +35,9 @@
      * @param body the request body. Must be an array of size 0 if the request
      * method is GET or DELETE.
      * @param userAgentString the user agent string to attach to all http communications
+     * @param proxySettings The proxy settings to use when connecting. If null then no proxy will be used
      */
-    public HttpsRequest(URL url, HttpsMethod method, byte[] body, String userAgentString)
+    public HttpsRequest(URL url, HttpsMethod method, byte[] body, String userAgentString, ProxySettings proxySettings)
```

Comment:
Please add a new method instead of modifying existing one.

Reply:
Sure thing, I can do that